koku/api/report/ocp/provider_map.py

```diff
@@ -1614,6 +1614,21 @@ def __init__(self, provider, report_type):
                             ),
                         },
                         "filter": [{"field": "data_source", "operation": "exact", "parameter": "Pod"}],
+                        "conditional_filter": {
```

Comment:
I think this should be called `conditional_exclude` instead of `filter`.

OR, a potentially better idea is to just call this `conditionals` so that way we can add `filters` as well without the naming seeming wrong.

Another thought here. The structure of this dict really limits which table we can create extra filters/exclusions for. We could make this a dict of dicts where the key is the table, and the value is a dict of `exclude`s or `filter`s. In this specific example, we could:
```
"conditionals": {
    OCPUsageLineItemDailySummary: {
        "exclude": [ {
                                    "field": "namespace",
                                    "operation": "exact",
                                    "parameter": "Workers Unallocated Capacity",
                                },
                                {
                                    "field": "namespace",
                                    "operation": "exact",
                                    "parameter": "Platform Unallocated Capacity",
                                }, ] ,
    },
}
```

Then in the `_provider_map_conditional_exclusions`, we could do:
```
exclusions = QueryFilterCollection()
to_exclude = self._mapper.report_type_map.get("conditional_filter", {}).get(self.query_table, {}).get("exclude", [])
for exclusion in to_exclude:
    exclusions.add(**exclusion)
return exclusions.compose()
```

Doing this eliminates a bunch of conditional checks.